File: Year02/2nd semester/MPP/Lab/mpp-proiect-java-AlexandruPaul21/JavaProjectNetworking/ProjectNetworking/src/main/java/org/example/dto/FlightDto.java
package org.example.dto;

import org.example.Flight;

import java.io.Serializable;

public class FlightDto implements Serializable {
    private Long id;
    private String destination;
    private String departure;
    private String airport;
    private String avbSeats;

    public FlightDto(Flight flight) {
        this.id = flight.getId();
        this.destination = flight.getDestination();
        this.departure = flight.getDepartureDateTime().toString();
        this.airport = flight.getAirport();
        this.avbSeats = String.valueOf(flight.getAvailableSeats());
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getDestination() {
        return destination;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public String getDeparture() {
        return departure;
    }

    public void setDeparture(String departure) {
        this.departure = departure;
    }

    public String getAirport() {
        return airport;
    }

    public void setAirport(String airport) {
        this.airport = airport;
    }

    public String getAvbSeats() {
        return avbSeats;
    }

    public void setAvbSeats(String avbSeats) {
        this.avbSeats = avbSeats;
    }
}

File: Year02/2nd semester/MPP/Lab/mpp-proiect-csharp-AlexandruPaul21/NetworkingProject/networking/FlightDto.cs
using model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace networking
{
    [Serializable]
    public class FlightDto
    {
        public long id { get; set; }
        public String destination { get; set; }
        public String departure { get; set; }
        public String airport { get; set; }
        public String avbSeats { get; set; }

        public FlightDto(Flight flight)
        {
            this.id = flight.Id;
            this.destination = flight.Destination;
            this.departure = flight.DepartureDateTime.ToString();
            this.airport = flight.Airport;
            this.avbSeats = flight.AvailableSeats.ToString();
        }
    }
}
